src/network/core_utils.rs

```diff
@@ -34,6 +35,68 @@ impl CoreUtils {
         final_slice.join(":")
     }
 
+    pub fn get_macvlan_mode_from_string(mode: &str) -> Result<u32, std::io::Error> {
+        // Replace to constant from library once this gets merged.
+        // TODO: use actual constants after https://github.com/little-dude/netlink/pull/200
+        match mode {
```

Comment:
Can we normalize the input to lowercase to avoid simple user errors?

Reply:
IMO The netavark format should be as strict as possible. I would prefer to normalize in podman if needed.

Reply:
Ack, I'm OK with this